include/spirv-tools/linker.hpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2017 Pierre Moreau
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SPIRV_TOOLS_LINKER_HPP_
+#define SPIRV_TOOLS_LINKER_HPP_
+
+#include <cstdint>
+
+#include <memory>
+#include <vector>
+
+#include "libspirv.hpp"
+
+namespace spvtools {
+
+class SpirvLinker {
+ public:
+  enum {
+    kDefaultLinkerOptions = SPV_LINKER_OPTION_NONE
+  };
+
+  // Constructs an instance targeting the given environment |env|.
+  //
+  // The constructed instance will have an empty message consumer, which just
+  // ignores all messages from the library. Use SetMessageConsumer() to supply
+  // one if messages are of concern.
+  explicit SpirvLinker(spv_target_env env);
+
+  // Disables copy/move constructor/assignment operations.
+  SpirvLinker(const SpirvLinker&) = delete;
+  SpirvLinker(SpirvLinker&&) = delete;
+  SpirvLinker& operator=(const SpirvLinker&) = delete;
+  SpirvLinker& operator=(SpirvLinker&&) = delete;
+
+  // Destructs this instance.
+  ~SpirvLinker();
+
+  // Sets the message consumer to the given |consumer|. The |consumer| will be
+  // invoked once for each message communicated from the library.
+  void SetMessageConsumer(MessageConsumer consumer);
+
+  bool Link(const std::vector<std::vector<uint32_t>>& binaries,
```

Comment:
Need method comment.  What is the return value?
For boolean return, we usually have true meaning success.